src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -118,4 +127,79 @@ public void validate(@Nullable Schema inputSchema) {
       }
     }
   }
+
+  private void validatePartitionProperties(@Nullable Schema schema) {
+    Table table = getDestinationTable();
+    if (table != null) {
```

Comment:
looks like we have 2 methods here instead of 1 with nullable parameter. The first one validates that user-defined field name matches column name from table definition and the second one validates field type for partitioning. 
1. we should split logic into 2 methods
2. shouldn't we check field time in first case?